### DISCUSSION TITLE
Why Hydration warn when I use pinia with useAsyncData

### DISCUSSION THREAD

**Discussion by Mini-ghost** [nuxt/nuxt#16019] [Questions]
After I get API response from axios, I save the data to `text` the pinia store state. And render in my `app.vue` template. When I use the return data of `useAsyncData`, that is work well. But When I use saved in pinia state, that will show me a warn message

```text
[Vue warn]: Hydration text content mismatch in <div>
```

Is this reasonable? Is pinia should not work with `useAsyncData` ?

This is my sample code

`store/app.ts`

```ts
import axios from 'axios';
import { defineStore } from 'pinia';

export const useAppStore = defineStore('app', {
  state: () => ({
    text: 'default'
  }),
  actions: {
    async GET_TEXT() {
      const res = await axios(API_URL).then(res => res.data)
      this.text = res
      return res;
    },
  },
});
```

`app.vue`

```vue
<template>
  <div>
    <div>
      {{ text }}
    </div>
  </div>
</template>

<script setup lang="ts">
import { storeToRefs } from 'pinia'
import { useAppStore } from '@/store/app';

const { text } = storeToRefs(useAppStore());
const { GET_TEXT } = useAppStore();
const { data } = useAsyncData('GET_TEXT', () => GET_TEXT(), {
  default: () => ‘’
});
</script>
```

**Comment by dpmango** [comment]
Seems to be working with `rc.11`

`store/product.js`

```js
export const useProductStore = defineStore('product', {
  state: () => {
    return {
      categories: [],
    }
  },
  actions: {
    async getCategories() {
      // useApi = $fetch wrapper
      const data = await useApi('catalog/get-categories', {
        method: 'GET',
      })

      this.categories = [...data]

      return data
    },
  },
})
```

`app.vue`

```vue
<script setup>
import { useProductStore } from '~/store'

const productStore = useProductStore()

await useAsyncData('categories', () => productStore.getCategories())
</script>
```

`Header.vue` and `Footer.vue`
```vue
<li v-for="link in productStore.categories">
  <NuxtLink :to="link.slug">{{ link.title }}</NuxtLink>
</li>
```

One catch up I was facing with Hydration warn (node missmatch) - unable to use `useAsyncData` inside one component, i.e. `Header.vue`, and share pinia state inside `Footer.vue`. Had to move request on app-level, probably will work on page-level as well

**Comment by brunodeangelis** [comment]
I'm having this same issue. What's the correct way to use Nuxt with Pinia without having hydration issues?

**Comment by martinapetrova** [comment]
Adding a `hydrate` function in my store solved this issue for me: https://pinia.vuejs.org/cookbook/composables.html#SSR
